### DISCUSSION TITLE
String Literals and Headers with HTMX

### DISCUSSION THREAD

**Discussion by zthacker** [a-h/templ#962] [Q&A] [answered]
Hey All,

I am having an issue passing a string into a hx-header

`templ Head(csrf string) {
            <div>
                <input id="vdocsearch" name="vaultdocsearch" type="text" class="flex justify-center"
                onkeyup="debounceSearch()"
                hx-post="/search"
                hx-trigger="keyup delay:500ms"
                hx-target="#results"
                hx-indicator=".htmx-indicator"
                hx-headers='{"X-CSRF-Token": { csrf }'
                class="p-2 border rounded w-full"
                <div id="results"></div>
            </div>

}`

The header isn't being populated and I am not sure why -- when I was reading the docs, it looks like this should work since the input is a string. Thoughts?

**Comment by pharrisee** [comment, marked as answer]
I'd personally change the `hx-headers` line to look something more like this:

```html
hx-headers={ fmt.Sprintf(`{"X-CSRF-Token":"%s"}`,csrf)}
```

String interpolation isn't supported in templ as far as I can see so you'll have to build the string yourself.
